src/sketch.lisp

```diff
@@ -106,13 +115,28 @@
 
 (defmethod initialize-instance :around ((instance sketch) &key &allow-other-keys)
   (initialize-sketch)
-  (call-next-method)
+  (sdl2:in-main-thread ()
+    (call-next-method))
   (kit.sdl2:start))
 
 (defmethod initialize-instance :after ((instance sketch) &rest initargs &key &allow-other-keys)
-  (initialize-environment instance)
   (apply #'prepare instance initargs)
-  (initialize-gl instance))
+  (setf (sketch-%window instance)
+        (make-instance 'sketch-window
+                       :title (sketch-title instance)
+                       :w (sketch-width instance)
+                       :h (sketch-height instance)
+                       :fullscreen (sketch-fullscreen instance)
+                       :resizable (if (sketch-resizable instance)
+                                      sdl2-ffi:+true+
+                                      sdl2-ffi:+false+)
+                       :sketch instance))
+  (initialize-environment instance)
+  (initialize-gl instance)
+  ;; These will have been added in the call to PREPARE.
+  (loop for f in (sketch-%delayed-init-funs instance)
+        do (funcall f))
```

Comment:
Just a suggestion (feel free to ignore): it might be more readable in functional style:
```lisp
(map nil #'funcall (sketch-%delayed-init-funs instance))
```

Reply:
I find the loop version more readable, haha.